'Error: parsing "": cannot parse an empty string' - `azurerm_site_recovery_replicated_vm`

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.4.5

### AzureRM Provider Version

3.63.0

### Affected Resource(s)/Data Source(s)

azurerm_site_recovery_replicated_vm

### Terraform Configuration Files

```hcl
resource "azurerm_site_recovery_replicated_vm" "vm_replication" {
  for_each                                  = { for k, v in merge(module.windows-server, module.windows-sql-server) : k => v if var.is_dr_enabled }
  name                                      = "rep-${each.value.virtual_machine.name}"
  resource_group_name                       = var.asr_rsv_resource_group_name
  recovery_vault_name                       = var.asr_rsv_name
  target_availability_set_id                = azurerm_availability_set.availability_set_dr[regex("(.*?)-\\d+$", each.key)[0]].id
  source_recovery_fabric_name               = var.asr_fabric_source_name
  source_vm_id                              = each.value.virtual_machine.id
  recovery_replication_policy_id            = var.asr_replication_policy_id
  source_recovery_protection_container_name = var.asr_protection_container_source_name

  target_resource_group_id                = var.dr_resource_group_id
  target_recovery_fabric_id               = var.asr_fabric_target_id
  target_recovery_protection_container_id = var.asr_protection_container_target_id

  dynamic "managed_disk" {
    for_each = each.value.managed_disks

    content {
      disk_id                    = managed_disk.value.id
      staging_storage_account_id = var.asr_cache_storage_account_source_id
      target_resource_group_id   = var.dr_resource_group_id
      target_disk_type           = managed_disk.value.storage_account_type
      target_replica_disk_type   = managed_disk.value.storage_account_type
    }
  }

  network_interface {
    source_network_interface_id = each.value.network_interface_id
    target_subnet_name          = reverse(split("/", each.value.subnet_id))[0]
  }

  provider = azurerm.dr
}
```


### Debug Output/Panic Output

```shell
2023-07-04T14:21:44.405+0100 [ERROR] provider.terraform-provider-azurerm_v3.63.0_x5: Response contains error diagnostic: @module=sdk.proto diagnostic_detail= diagnostic_severity=ERROR tf_proto_version=5.3 tf_provider_addr=provider @caller=github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/internal/diag/diagnostics.go:55 tf_req_id=70c4a26f-8a37-0288-c5e5-e7a0c6c8866a tf_resource_type=azurerm_site_recovery_replicated_vm diagnostic_summary="parsing "": cannot parse an empty string" tf_rpc=ReadResource timestamp=2023-07-04T14:21:44.405+0100
2023-07-04T14:21:44.406+0100 [ERROR] vertex "module.main.azurerm_site_recovery_replicated_vm.vm_replication[\"vm-1\"]" error: parsing "": cannot parse an empty string
```


### Expected Behaviour

Successful preparation of the plan

### Actual Behaviour

After azurerm upgrade from version 3.62.1 to 3.63.0, terraform plan starts failing for `azurerm_site_recovery_replicated_vm` with following error:
```
│ Error: parsing "": cannot parse an empty string
│ 
│   with module.main.azurerm_site_recovery_replicated_vm.vm_replication["vm-1"],
│   on ../../main/site_recovery.tf line 14, in resource "azurerm_site_recovery_replicated_vm" "vm_replication":
│   14: resource "azurerm_site_recovery_replicated_vm" "vm_replication" {
│ 
```
I checked debug log (see above), this information doesn't tell me what exactly is wrong. I checked all resource inputs via passing to terraform output, and there is no empty string.
I didn't make any changes to the Terraform code. It starts working after downgrading to 3.62.1

### Steps to Reproduce

`terraform plan`

### Important Factoids

_No response_

### References

_No response_

--- a/internal/services/recoveryservices/site_recovery_replicated_vm_resource.go
+++ b/internal/services/recoveryservices/site_recovery_replicated_vm_resource.go
@@ -754,8 +754,8 @@ func resourceSiteRecoveryReplicatedItemRead(d *pluginsdk.ResourceData, meta inte
 			d.Set("target_resource_group_id", recoveryGroupId)
 
 			availabilitySetId := ""
-			if id := pointer.From(a2aDetails.RecoveryAvailabilitySet); id != "" {
-				parsedAvailabilitySetId, err := availabilitysets.ParseAvailabilitySetIDInsensitively(availabilitySetId)
+			if respAvailabilitySetId := pointer.From(a2aDetails.RecoveryAvailabilitySet); respAvailabilitySetId != "" {
+				parsedAvailabilitySetId, err := availabilitysets.ParseAvailabilitySetIDInsensitively(respAvailabilitySetId)
 				if err != nil {
 					return err
 				}
@@ -764,8 +764,8 @@ func resourceSiteRecoveryReplicatedItemRead(d *pluginsdk.ResourceData, meta inte
 			d.Set("target_availability_set_id", availabilitySetId)
 
 			targetNetworkId := ""
-			if id := pointer.From(a2aDetails.SelectedRecoveryAzureNetworkId); id != "" {
-				parsedTargetNetworkId, err := commonids.ParseVirtualNetworkIDInsensitively(id)
+			if respTargetNetworkId := pointer.From(a2aDetails.SelectedRecoveryAzureNetworkId); respTargetNetworkId != "" {
+				parsedTargetNetworkId, err := commonids.ParseVirtualNetworkIDInsensitively(respTargetNetworkId)
 				if err != nil {
 					return err
 				}
@@ -774,8 +774,8 @@ func resourceSiteRecoveryReplicatedItemRead(d *pluginsdk.ResourceData, meta inte
 			d.Set("target_network_id", targetNetworkId)
 
 			testNetworkId := ""
-			if tfoNetworkId := pointer.From(a2aDetails.SelectedTfoAzureNetworkId); tfoNetworkId != "" {
-				parsedTfoNetworkId, err := commonids.ParseVirtualNetworkIDInsensitively(tfoNetworkId)
+			if respTfoNetworkId := pointer.From(a2aDetails.SelectedTfoAzureNetworkId); respTfoNetworkId != "" {
+				parsedTfoNetworkId, err := commonids.ParseVirtualNetworkIDInsensitively(respTfoNetworkId)
 				if err != nil {
 					return err
 				}
@@ -784,8 +784,8 @@ func resourceSiteRecoveryReplicatedItemRead(d *pluginsdk.ResourceData, meta inte
 			d.Set("test_network_id", testNetworkId)
 
 			proximityPlacementGroupId := ""
-			if id := pointer.From(a2aDetails.RecoveryProximityPlacementGroupId); id != "" {
-				parsedProximityPlacementGroupId, err := proximityplacementgroups.ParseProximityPlacementGroupIDInsensitively(id)
+			if respProximityPlacementGroupId := pointer.From(a2aDetails.RecoveryProximityPlacementGroupId); respProximityPlacementGroupId != "" {
+				parsedProximityPlacementGroupId, err := proximityplacementgroups.ParseProximityPlacementGroupIDInsensitively(respProximityPlacementGroupId)
 				if err != nil {
 					return err
 				}
@@ -794,8 +794,8 @@ func resourceSiteRecoveryReplicatedItemRead(d *pluginsdk.ResourceData, meta inte
 			d.Set("target_proximity_placement_group_id", proximityPlacementGroupId)
 
 			recoveryBootDiagStorageAccount := ""
-			if id := pointer.From(a2aDetails.RecoveryBootDiagStorageAccountId); id != "" {
-				parsedRecoveryBootDiagStorageAccount, err := storageaccounts.ParseStorageAccountIDInsensitively(id)
+			if respBootDiagStorageAccountId := pointer.From(a2aDetails.RecoveryBootDiagStorageAccountId); respBootDiagStorageAccountId != "" {
+				parsedRecoveryBootDiagStorageAccount, err := storageaccounts.ParseStorageAccountIDInsensitively(respBootDiagStorageAccountId)
 				if err != nil {
 					return err
 				}
@@ -804,8 +804,8 @@ func resourceSiteRecoveryReplicatedItemRead(d *pluginsdk.ResourceData, meta inte
 			d.Set("target_boot_diagnostic_storage_account_id", recoveryBootDiagStorageAccount)
 
 			capReservationGroupId := ""
-			if id := pointer.From(a2aDetails.RecoveryCapacityReservationGroupId); id != "" {
-				parsedCapReservaGroupId, err := capacityreservationgroups.ParseCapacityReservationGroupIDInsensitively(id)
+			if respCapacityGroupId := pointer.From(a2aDetails.RecoveryCapacityReservationGroupId); respCapacityGroupId != "" {
+				parsedCapReservaGroupId, err := capacityreservationgroups.ParseCapacityReservationGroupIDInsensitively(respCapacityGroupId)
 				if err != nil {
 					return err
 				}
@@ -814,8 +814,8 @@ func resourceSiteRecoveryReplicatedItemRead(d *pluginsdk.ResourceData, meta inte
 			d.Set("target_capacity_reservation_group_id", capReservationGroupId)
 
 			vmssId := ""
-			if id := pointer.From(a2aDetails.RecoveryVirtualMachineScaleSetId); id != "" {
-				parsedVmssId, err := computeParse.VirtualMachineScaleSetIDInsensitively(id)
+			if respVmssId := pointer.From(a2aDetails.RecoveryVirtualMachineScaleSetId); respVmssId != "" {
+				parsedVmssId, err := computeParse.VirtualMachineScaleSetIDInsensitively(respVmssId)
 				if err != nil {
 					return err
 				}

--- a/internal/services/recoveryservices/site_recovery_replicated_vm_resource_test.go
+++ b/internal/services/recoveryservices/site_recovery_replicated_vm_resource_test.go
@@ -197,6 +197,21 @@ func TestAccSiteRecoveryReplicatedVm_targetDiskEncryption(t *testing.T) {
 	})
 }
 
+func TestAccSiteRecoveryReplicatedVm_withAvailabilitySet(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_site_recovery_replicated_vm", "test")
+	r := SiteRecoveryReplicatedVmResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.withAvailabilitySet(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
 func (SiteRecoveryReplicatedVmResource) template(data acceptance.TestData) string {
 	tags := ""
 	if strings.HasPrefix(strings.ToLower(data.Client().SubscriptionID), "85b3dbca") {
@@ -1917,6 +1932,53 @@ resource "azurerm_site_recovery_replicated_vm" "test" {
 }
 `, r.template(data), data.RandomInteger)
 
+}
+
+func (r SiteRecoveryReplicatedVmResource) withAvailabilitySet(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_availability_set" "test" {
+  name                = "acctestavset-%d"
+  location            = azurerm_resource_group.test2.location
+  resource_group_name = azurerm_resource_group.test2.name
+  managed             = true
+}
+
+resource "azurerm_site_recovery_replicated_vm" "test" {
+  name                                      = "repl-%[2]d"
+  resource_group_name                       = azurerm_resource_group.test2.name
+  recovery_vault_name                       = azurerm_recovery_services_vault.test.name
+  source_vm_id                              = azurerm_virtual_machine.test.id
+  source_recovery_fabric_name               = azurerm_site_recovery_fabric.test1.name
+  recovery_replication_policy_id            = azurerm_site_recovery_replication_policy.test.id
+  source_recovery_protection_container_name = azurerm_site_recovery_protection_container.test1.name
+
+  target_availability_set_id              = azurerm_availability_set.test.id
+  target_resource_group_id                = azurerm_resource_group.test2.id
+  target_recovery_fabric_id               = azurerm_site_recovery_fabric.test2.id
+  target_recovery_protection_container_id = azurerm_site_recovery_protection_container.test2.id
+
+  managed_disk {
+    disk_id                    = azurerm_virtual_machine.test.storage_os_disk[0].managed_disk_id
+    staging_storage_account_id = azurerm_storage_account.test.id
+    target_resource_group_id   = azurerm_resource_group.test2.id
+    target_disk_type           = "Premium_LRS"
+    target_replica_disk_type   = "Premium_LRS"
+  }
+
+  network_interface {
+    source_network_interface_id   = azurerm_network_interface.test.id
+    target_subnet_name            = azurerm_subnet.test2.name
+    recovery_public_ip_address_id = azurerm_public_ip.test-recovery.id
+  }
+
+  depends_on = [
+    azurerm_site_recovery_protection_container_mapping.test,
+    azurerm_site_recovery_network_mapping.test,
+  ]
+}
+`, r.template(data), data.RandomInteger)
 }
 
 func (r SiteRecoveryReplicatedVmResource) Exists(ctx context.Context, clients *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {

